src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,337 @@
+from qt3utils.errors import  QT3Error
+from typing import Any, Tuple, List, Union
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+try:
+    import clr
+
+    lf_root = Path(os.environ['LIGHTFIELD_ROOT'])
+    automation_path = lf_root / 'PrincetonInstruments.LightField.AutomationV4.dll'
+    addin_path = lf_root / 'AddInViews' / 'PrincetonInstruments.LightFieldViewV4.dll'
+    support_path = lf_root / 'PrincetonInstruments.LightFieldAddInSupportServices.dll'
+
+    addin_class = clr.AddReference(str(addin_path))
+    automation_class = clr.AddReference(str(automation_path))
+    support_class = clr.AddReference(str(support_path))
+
+    import PrincetonInstruments.LightField as lf
+
+    clr.AddReference("System.Collections")
+    clr.AddReference("System.IO")
+    from System.Collections.Generic import List
+    from System import String, Int32, Int64, IntPtr, Double
+    from System.IO import FileAccess
+except KeyError as e:
+    logger.error(f"KeyError {e} during import")
+except ImportError as e:
+    logger.error(f"Unable to import packages: {e}")
+
+
+class LightfieldApplicationManager:
+    def __init__(self, visible: bool) -> None:
+        self._automation = lf.Automation.Automation(visible, List[String]())
+        self._application = self._automation.LightFieldApplication
+        self._experiment = self._application.Experiment
+
+    @property
+    def experiment(self) -> Any:
+        return self._experiment
+
+    def __del__(self) -> None:
+        """
+        Uses Python garbage collection method used to terminate
+        AddInProcess.exe, if this is not done, LightField will not reopen
+        properly.
+        """
+        #NOTE: May need to call '.Dispose()' here again as qt3scan may fail to call 'close'
+        self._automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
+
+    def set(self, setting: str, value: Any) -> None:
+        """
+        Helper function for setting experiment parameters with basic value
+        checking.
+        """
+        if self.experiment.Exists(setting):
+            if self.experiment.IsValid(setting, value):
+                self.experiment.SetValue(setting, value)
+            else:
+                # TODO: might need to add a proper exception
+                logger.error(f'Invalid value: {value} for setting: {setting}.')
+        else:
+            # TODO: might need to add a proper exception
+            logger.error(f'Invalid setting: {setting}.')
+
+    def get(self, setting: Any) -> Any:
+        """
+        Helper function for getting experiment parameters with a basic check
+        to see if a specific setting is valid.
+        """
+        if self.experiment.Exists(setting):
+            value = self.experiment.GetValue(setting)
+        else:
+            value = []
+            logger.error(f'Invalid setting: {setting}.')
+        return value
+
+    def load_experiment(self, value: str) -> None:
```

Comment:
rename "value" argument to better reflect what it is. Also add a docstring explaining what it means.

Reply:
it seems that common usage with @property.setter functions is to name the variable passed in to 'value'. this is not a property.setter, but I suppose that was the spirit of this function. `experiment_name` might work here. @cmordi does LightField require that you provide a full path to an experiment file, or does it just look in one default location?

Reply:
I do not think we should use `experiment_name`, just from an aesthetic point of view, it has been used too many times so people might get confused. I can think of another name. 

To answer your question @gadamc, no. Lightfield does not require that I provide a full path to an experiment file. It just looks in a default location